src/core/src/op/scatter_update.cpp

```diff
@@ -2,134 +2,116 @@
     : util::ScatterBase(data, indices, updates, axis) {}
 
-shared_ptr<Node> op::v3::ScatterUpdate::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> ScatterUpdate::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v3_ScatterUpdate_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<v3::ScatterUpdate>(new_args.at(0), new_args.at(1), new_args.at(2), new_args.at(3));
-}
-
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace scatter_update {
-namespace {
-template <element::Type_t ET>
-std::vector<int64_t> get_indices(const HostTensorPtr& in) {
-    auto data_ptr = in->get_data_ptr<ET>();
-    return std::vector<int64_t>(data_ptr, data_ptr + in->get_element_count());
+    return std::make_shared<ScatterUpdate>(new_args.at(0), new_args.at(1), new_args.at(2), new_args.at(3));
 }
-}  // namespace
-}  // namespace scatter_update
 
-#define GET_INDICES(a, ...)                                                                   \
-    case element::Type_t::a: {                                                                \
-        OV_OP_SCOPE(OV_PP_CAT3(get_scatter_update_indices, _, a));                            \
-        indices_casted_vector = scatter_update::get_indices<element::Type_t::a>(__VA_ARGS__); \
-    } break;
+bool ScatterUpdate::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+    OV_OP_SCOPE(v3_ScatterUpdate_evaluate);
+    OPENVINO_ASSERT(inputs.size() == 4);
+    OPENVINO_ASSERT(outputs.size() == 1);
 
-bool op::v3::ScatterUpdate::evaluate_scatter_update(const HostTensorVector& outputs,
-                                                    const HostTensorVector& inputs) const {
     const auto& data = inputs[0];
     const auto& indices = inputs[1];
     const auto& updates = inputs[2];
     const auto& axis = inputs[3];
-    const auto& out = outputs[0];
-
-    const auto elem_size = data->get_element_type().size();
-    out->set_shape(data->get_shape());
-
-    OPENVINO_ASSERT(axis->get_element_type().is_integral_number(), "axis element type is not integral data type");
-
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    int64_t axis_val = host_tensor_2_vector<int64_t>(axis)[0];
-    if (axis_val < 0) {
-        axis_val = ngraph::normalize_axis(this, axis_val, static_cast<int64_t>(data->get_shape().size()));
-    }
-    OPENVINO_SUPPRESS_DEPRECATED_END
-
-    std::vector<int64_t> indices_casted_vector;
-    switch (indices->get_element_type()) {
-        GET_INDICES(i8, indices);
-        GET_INDICES(i16, indices);
-        GET_INDICES(i32, indices);
-        GET_INDICES(i64, indices);
-        GET_INDICES(u8, indices);
-        GET_INDICES(u16, indices);
-        GET_INDICES(u32, indices);
-        GET_INDICES(u64, indices);
+    auto& output = outputs[0];
+
+    OPENVINO_ASSERT(axis.get_element_type().is_integral_number(), "axis element type is not integral data type");
+
+    switch (indices.get_element_type()) {
```

Comment:
should it be replaced with `IfTypeOf`? I see `GET_INDICES` also enables CC, so use of structure will also help to restore CC (via `IF_TYPE_OF` macro)